scripts/west_commands/patch.py

```diff
@@ -0,0 +1,327 @@
+                apply-command: git apply
+                comments: |
+                  Songs about rainbows - why are there so many??
+                custom:
+                  possible-muppets-to-ask-for-clarification-with-the-above-question:
+                    - Miss Piggy
+                    - Gonzo
+                    - Fozzie Bear
+                    - Animal
+            """),
+        )
+
+        parser.add_argument(
+            "-b",
+            "--patch-base",
+            help="Directory containing patch files",
+            metavar="DIR",
+            default=Path("WEST_MANIFEST_DIR") / WEST_PATCH_BASE,
+        )
+        parser.add_argument(
+            "-l",
+            "--patch-yml",
+            help="Path to patches.yml file",
+            metavar="FILE",
+            default=Path("WEST_MANIFEST_DIR") / WEST_PATCH_YAML,
+        )
+        parser.add_argument(
+            "-w",
+            "--west-workspace",
+            help="West workspace",
+            metavar="DIR",
+            default=Path("WEST_TOPDIR"),
+        )
+
+        subparsers = parser.add_subparsers(
+            dest="subcommand",
+            metavar="<subcommand>",
+            help="select a subcommand. If omitted treat it as 'list'",
+        )
+
+        apply_arg_parser = subparsers.add_parser(
+            "apply",
+            help="Apply patches",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Applying Patches:
+
+                Run "west patch apply" to apply patches.
+            """
+            ),
+        )
+        apply_arg_parser.add_argument(
+            "-r",
+            "--roll-back",
+            help="Roll back if any patch fails to apply",
+            action="store_true",
+            default=False,
+        )
+
+        subparsers.add_parser(
+            "clean",
+            help="Clean patches",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Cleaning Patches:
+
+                Run "west patch clean" to clean patches.
+            """
+            ),
+        )
+
+        subparsers.add_parser(
+            "list",
+            help="List patches",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Listing Patches:
+
+                Run "west patch list" to list patches.
+            """
+            ),
+        )
+
+        return parser
+
+    def filter_args(self, args):
+        try:
+            manifest_path = self.config.get("manifest.path")
+        except BaseException:
+            self.die("could not retrieve manifest path from west configuration")
+
+        topdir = Path(self.topdir)
+        manifest_dir = topdir / manifest_path
+        _WEST_MANIFEST_DIR = Path("WEST_MANIFEST_DIR")
+        _WEST_TOPDIR = Path("WEST_TOPDIR")
```

Comment:
I suggested the underscore prefix and uppercase to put these as "private" scoped inside this module, sorry didn't make that clear in the previous comment.

Reply:
Makes sense. Do you want me to change it again?

Reply:
I'd move these to the top of the file, so you can use them in for the `parser` too.

Reply:
Fixed!